`go get golang.org/x/lint/golint` is failing in CI

When raising a new PR, e.g. https://github.com/elastic/elastic-agent-client/pull/103

we see the following failure ([bk link](https://buildkite.com/elastic/elastic-agent-client/builds/134)):

```
# golang.org/x/tools/go/types/objectpath
/go/src/golang.org/x/tools/go/types/objectpath/objectpath.go:397:10: meth.Origin undefined (type *types.Func has no field or method Origin)
Error: running "go get golang.org/x/lint/golint" failed with exit code 2
🚨 Error: The command exited with status 2
user command error: exit status 2
```

--- a/.buildkite/pipeline.yml
+++ b/.buildkite/pipeline.yml
@@ -1,6 +1,6 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
 env:
-  SETUP_GOLANG_VERSION: '1.18.10'
+  SETUP_GOLANG_VERSION: '1.21.8'
 
 steps:
   - label: "Update"

